### ISSUE TITLE
How to configure used namespace (non-default) for i18n translations?

### ISSUE THREAD

**Issue by jani-pernu** [ghiscoding/aurelia-slickgrid#338]
How to tell the grid which namespace to use for i18n translations?

This setup works in our project. The translations from aurelia-slickgrid.json are displayed in the grid.
```
return instance.setup({
	attributes: ["t", "i18n"],
	backend: {
		loadPath: "/locales/{{lng}}/{{ns}}.json"
	},
	debug: true,
	defaultNS: "aurelia-slickgrid",
	fallbackLng: "fi",
	lng: "fi",
	ns: [
		"common",
		"aurelia-slickgrid"
	]
});
```

The problem is that we already have utilized a different default namespace (common).
So the setup above is not possible and we have to use the one below:
```
return instance.setup({
	attributes: ["t", "i18n"],
	backend: {
		loadPath: "/locales/{{lng}}/{{ns}}.json"
	},
	debug: true,
	defaultNS: "common",
	fallbackLng: "fi",
	lng: "fi",
	ns: [
		"common",
		"aurelia-slickgrid",
		"other-namespaces..."
	]
});
```

1. So how I configure the grid to utilize "aurelia-slickgrid" namespace?
Didn't find any good candidate from interface GridOption.

Now getting a lot of missing key prints in console and the translations are not displayed.
"i18next::translator: missingKey fi common CONTAINS CONTAINS"

2. Also tried to copy all the translations from aurelia-slickgrid.json to common.json but that didn't help either. Still getting those missing key prints. This should work?

3. Moreover getting those missing key prints still after successfully applying the custom locale approach without enableTranslate in grid options. The grid shouldn't look for translations if they are not in use at all.

| Software          | Version(s) |
| ----------------- | ---------- |
| Aurelia           | 0.6.0
| Aurelia-Slickgrid | 2.17.9
| TypeScript        | 3.8.2
| Operating System  | Win10
| NPM/Node/Yarn     | 6.13.4/13.5.0.

**Comment by ghiscoding** [comment]
Which version of I18N are you using? My lib is still requiring the same version number `"aurelia-i18n": "^2.3.2"` because I think the newer `3.x` version would require a major version bump in my lib and I don't really want at this point. 

I'm not doing anything special with I18N, you can see the config I have in the [Aurelia-Slickgrid-Demo](https://github.com/ghiscoding/aurelia-slickgrid-demos/blob/master/webpack-bs3-demo/src/main.ts)

**Comment by jani-pernu** [comment]
I now downloaded aurelia-slickgrid repo locally and looked through some code.
E.g. compoundInputFilter.ts row 176: **this.i18n.tr('CONTAINS')**
Grid isn't using any namespace prefix when looking up the translations i.e. the translations are assumed to be in the default namespace. This is problematic when applying grid to existing project that already has many translations in the default namespace. The existing translation keys could collide with the ones defined by the grid.

So this issue should be a feature request then.

1.  Need a way to tell the grid which namespace prefix to use for translations.
```
this.gridOptions = {
	translationNamespace: "aurelia-slickgrid"
}
```
3. Grid should then utilize this namespace when looking up the translations
```
const prefix = this.gridOptions.translationNamespace ? this.gridOptions.translationNamespace + ":" : "";
this.i18n.tr(prefix + "CONTAINS")
```

**Comment by ghiscoding** [comment]
ah ok I now understand your question, that is correct I'm not using namespace since I didn't think it was necessary for simple words. I'd be happy to receive a Pull Request for this enhancement

**Comment by ghiscoding** [comment]
@jani-pernu could you please review PR #344

**Comment by ghiscoding** [comment]
This is now available in latest version [2.18.x](https://github.com/ghiscoding/aurelia-slickgrid/releases). 

Please upvote ⭐ if you haven't already. 
Cheers
